src/core/dense_set.cc

```diff
@@ -179,16 +179,17 @@ void* DenseSet::PopDataFront(DenseSet::ChainVectorIterator it) {
   return ret;
 }
 
-void DenseSet::ClearInternal() {
+uint32_t DenseSet::ClearInternal(uint32_t start, uint32_t limit) {
   constexpr unsigned kArrLen = 32;
   ClearItem arr[kArrLen];
   unsigned len = 0;
 
-  for (auto it = entries_.begin(); it != entries_.end(); ++it) {
-    if (it->IsEmpty())
+  size_t end = min<size_t>(entries_.size(), start + limit);
```

Comment:
you mean `limit - start` here right ? Cause otherwise: 

`entries_.size()=10, start=3, limit=8` then `end=10` and you will iterate from `start=3` up to `end=10` when you wanted to start from `3` and end up to `limit=8`. 

Is there something I am missing here?

Reply:
unless you mean `limit` as `count` that is, how many elements from `start` you want to delete. (cause in my mind limit is more of like an index limit rather than the total elements you want to delete starting at pos `start`

Reply:
I meant count, not limit

Reply:
uh gotcha